qutip/correlation.py

```diff
@@ -1297,20 +1369,20 @@ def _correlation_mc_2t(H, state0, tlist, taulist, c_ops, a_op, b_op, c_op,
 
 # pseudo-inverse solvers
 
-def _spectrum_pi(H, wlist, c_ops, a_op, b_op, use_pinv=False):
+def _spectrum_pi(H, wlist, collapse_ops, A, B, use_pinv=False):
     """
     Internal function for calculating the spectrum of the correlation function
     :math:`\left<A(\\tau)B(0)\\right>`.
     """
 
-    L = H if issuper(H) else liouvillian(H, c_ops)
+    L = H if issuper(H) else liouvillian(H, collapse_ops)
 
     tr_mat = tensor([qeye(n) for n in L.dims[0][0]])
     N = np.prod(L.dims[0][0])
 
     A = L.full()
-    b = spre(b_op).full()
-    a = spre(a_op).full()
+    b = spre(B).full()
+    a = spre(A).full()
```

Comment:
`A` is a dense array when passed to `spre`.  This is the error.  It should be sparse, and then the result of `spre` is converted to dense.

Reply:
This is deeply unusual since I didn't touch that part of the code and spre has an if statement to throw an error when input is not a Qobj. 

Any idea how this happened?